hudi-client/src/test/java/org/apache/hudi/table/action/commit/TestUpsertPartitioner.java

```diff
@@ -252,8 +250,27 @@ public void testUpsertPartitionerWithSmallInsertHandling() throws Exception {
     assertEquals(BucketType.INSERT, partitioner.getBucketInfo(2).bucketType,
         "Bucket 2 is INSERT");
     assertEquals(3, insertBuckets.size(), "Total of 3 insert buckets");
-    assertEquals(0, insertBuckets.get(0).bucketNumber, "First insert bucket must be same as update bucket");
-    assertEquals(0.5, insertBuckets.get(0).weight, 0.01, "First insert bucket should have weight 0.5");
+
+    assertEquals(0, insertBuckets.get(0).getKey().bucketNumber,
+        "First insert bucket must be same as update bucket");
```

Comment:
lets avoid doing one at a time w/ hardcoded expected values. I mean, we should try to do in a loop. Can we build a map of <index -> ExpectedValue>. ExpectedValue should contain bucketNo, bucketType, weight, cumulativeWeight. Once we have this, these 20 lines of assertions will be folded to just 4 to 5 lines.

Reply:
got it.